MFAssignR/R/RecalFunctions.R

```diff
@@ -764,15 +740,33 @@ RecalList <- function(df) {
   Recal$Max <- round(Recal$Max, 3)
   Recal$SerScor <- ((Recal$Max - Recal$Min) / 14 + 1) / Recal$number.y
   Recal <- tidyr::unite(Recal, Range, Min, Max, sep = "-")
-  Recal <- Recal[-c(5, 7, 8, 9, 10, 11, 12)]
-  names(Recal)[3] <- "Series Index"
-  names(Recal)[2] <- "Number Observed"
-  names(Recal)[4] <- "Mass Range"
-  names(Recal)[5] <- "Tall Peak"
-  names(Recal)[6] <- "Abundance Score"
-  names(Recal)[9] <- "Series Score"
-  names(Recal)[8] <- "Peak Distance"
-  names(Recal)[7] <- "Peak Score"
+
+  columns_to_drop <- c(5, 7, 8, 9, 10, 11, 12)
+  Recal <- Recal[, -columns_to_drop]
+
+  columns_to_rename <- c(2, 3, 4, 5, 6, 7, 8, 9)
+  new_column_names <- c("Number Observed", "Series Index", "Mass Range", "Tall Peak", "Abundance Score", "Peak Score", "Peak Distance", "Series Score")
+  names(Recal)[columns_to_rename] <- new_column_names
   Recal <- Recal[!duplicated(Recal), ]
-  Recal
+}
+
+AddCalculatedSummary <- function(Recal) {
+  Recal <- dplyr::group_by(Recal, Index)
+  Recal <- dplyr::mutate(Recal,
+    Min = min(Exp_mass),
+    Max = max(Exp_mass),
+    MInt = mean(Abundance),
+    Maxmass = ifelse(Abundance == max(Abundance),
+      Exp_mass, NA
+    ), Maxint = max(Abundance),
+    Secint = sort(Abundance, TRUE)[2], Secmass = ifelse(Abundance == sort(Abundance, TRUE)[2], Exp_mass, NA)
+  )
+  return(Recal)
+}

```

Comment:
Please add documentation to the function to explain what it does.

Reply:
Added documentation [bb8267f](https://github.com/RECETOX/MFAssignR/pull/33/commits/bb8267fbe1d5fad2e670ea39860c006c47a8b540)